packages/react/src/components/navigation/FlowStepper/index.tsx

```diff
@@ -69,9 +84,27 @@ function FlowStepper<ExtraProps>({
   renderChildren,
   children,
 }: Props<ExtraProps>) {
-  const steps = React.Children.map(children, (child) => child.props.label);
-  const innerJSX = React.Children.map(children, (child, index) =>
-    index === activeIndex ? child : null,
+  const { steps, innerContents } = React.Children.toArray(children).reduce<{
+    steps: string[];
+    innerContents: React.ReactNode | null;
+  }>(
+    (acc, child, idx) => {
+      const index = (isElement(child) && child.props.index) ?? idx;
+      const label = isElement(child) && child.props.label;
+      const hidden = isElement(child) && child.props.hidden;
+
+      if (label && !hidden) {
```

Comment:
can't this maybe end up overriding another step if we only pass the index for one of them?

Reply:
the consumer of the lib should be responsible of passing sensible props in this case

Reply:
If this is case, I think we should maybe document that if we want to use Index we should probably use Index for **all** the other steps too. Can we maybe even enforce it using typescript?

Reply:
Anyway, approving as this would be mainly improvements.